### ISSUE TITLE
[BUG] Token Overlay switching attributes and not displaying

### ISSUE THREAD

**Issue by dcoughler** [Moerill/token-mold#145]
I have multiple worlds on my server, and the token overlay only appears to be working on some of them. On one of the ones that is failing, there are two symptoms.

1) If I select _ac.value [character, npc]_ and save, when I edit the overlay config the attribute changes to _ac.value [vehicle]_. 
DEBUGGING:
- I have removed the attribute and re-added it multiple times.
- I have also changed the order of the attributes. No luck.

2) The overlay does not appear.
DEBUGGING:
- I've matched the overlay settings to one of the working worlds.
- I've made sure the two worlds have all of the same modules enabled.

CONSOLE LOG:
Foundry VTT | Rendering ImageHoverHUD
VM29738:74 Uncaught TypeError: Cannot read properties of null (reading 'querySelectorAll')
[No packages detected]
    at selector (<anonymous>:74:43)
    at selector (<anonymous>:82:21)
    at selector (<anonymous>:82:21)
    at HTMLDocument.<anonymous> (<anonymous>:14:39)
selector @ VM29738:74
selector @ VM29738:82
selector @ VM29738:82
(anonymous) @ VM29738:14
18foundry.js:3053 Foundry VTT | Rendering ImageHoverHUD
foundry.js:3053 Foundry VTT | Rendering TokenMoldForm

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - FVTT Version: 9.269
 - Module version: 2.15.4
 - Host: Oracle Cloud (Linux)
 - System: DnD5e v.1.6.3

```
Active Modules:
----------
Ace Library v;
Actually Private Rolls v;
Always HP v;
Automatic Journal Icon Numbers v;
Better Rolls for 5e v;
Beyond20 companion module v;
Character Actions List dnd5e v;
Combat Focus v;
Combat Utility Belt v;
Compact DnDBeyond 5e Character Sheet v;
Compendium Folders v;
Cursor Hider v;
D&D Beyond Importer v;
DFreds Droppables v;
Dice Tray v;
DMDave - Mimic Museum v;
Doug's D&D Beyond Import 1 - Actors, Spells, and Items v;
Doug's Macros and Journals v;
Dynamic effects using Active Effects v;
Everybody Look v;
FXMaster v;
GM Screen v;
Image Hover v;
lib - Color Settings v;
libWrapper v0;
Macro Editor v;
Magic Items v;
Mana's Compendium Importer v;
Monk's Active Tile Triggers v;
Monk's TokenBar v;
Multilevel Tokens v;
Multiple Wall Point Mover (MWPM) v;
Party Resources v;
Perfect Vision v;
Pings v;
PopOut! v;
Popout Resizer v;
Quick Roll To Chat v;
✋ Raise My Hand v;
Safety Tools v;
Sane Magical Item Prices Compendium v;
Settings Extender v;
Shared Vision v;
Simplefog - Manual Fog of War v;
socketlib v;
The Tomb of Horrors v;
Tidy5e Sheet v;
Tidy UI - Game Settings v;
Token Action HUD v;
Tokenizer v;
Token Magic FX v;
Token Mold v;
Trigger Happy v;
Universal Battlemap Importer v;
Wooden UI v;
```

The first symptom may be a totally separate issue, but given that it also only occurs on the malfunctioning worlds, I though I should include it in case it gives you somewhere to look.

**Comment by Geekswordsman** [comment]
This issue with AC "resetting" (it's not, to be explained after) I am looking to resolve in v3; but rest assured even if it looks like it's resetting, it's really not.  

The drop downs for the overlay are automated based on attributes for each actor type in the system.  All actor types in dnd5e use the same attribute path for the AC value; PCs and NPCs share the same base template data whereas vehicles are separated and thus it gets a separate entry in the drop down.  Currently the overlay only stores the data path, not the document type that the overlay attribute is for.  Since vehicle is the first entry in the drop down list... it gets selected.  _But_ again, since it's the same data path, it will work regardless.

I can't replicate the second portion of this ticket.  I would recommend attempting to replicate with no other modules enabled; it's quite possible one of the other modules are interfering with the overlay.

**Comment by dyenrac** [comment]
I have the same issue. I found that the overlay works in Foundry run on my computer but not from my Foundry server in Forge.
I tried chrome, Firefox, Edge. I tried disabling all the other mods and it still did not work. I found that the icon images are distorted in the drop down but look right after one is selected.

**Comment by dyenrac** [comment]
I also noticed that whenever I select ac.value character, npc it changes to ac.value vehicle after I select save.
![Untitled](https://user-images.githubusercontent.com/117706154/200477234-3946e1d8-9eaf-4e4c-9c02-dee25014a0ed.jpg)

**Comment by Geekswordsman** [comment]
See my comment just before your report for an explanation about the AC swapping - the actual ac is going to be displayed.

I have not been able to duplicate the missing overlay on my local development instances, however I am re-writing Token Mold so it's possible any bug with the display will be fixed then.

**Comment by dyenrac** [comment]
> See my comment just before your report for an explanation about the AC swapping - the actual ac is going to be displayed.
> 
> I have not been able to duplicate the missing overlay on my local development instances, however I am re-writing Token Mold so it's possible any bug with the display will be fixed then.

**Comment by dyenrac** [comment]
I was running version 9  on Forge and version 10 on my PC to test. After I updated Foundry to version 10 the overlay started to work. I found that as the GM I can see the overlay for all PC's and NPC's. As a player  I can only see the  overlay on my own token. As the GM I need to see all the overlay for all token but it would be nice to have the option  to allow the players to see the overlay for all tokens, all PC tokens but not NPC's tokens, or for non of the token's.
